src/main/java/com/intellij/rt/debugger/agent/SharedFlowTransformer.java

```diff
@@ -58,16 +64,26 @@ private fun wrapInternalDebuggerCapture(value: Any?): Any = FlowValueWrapperInte
         ```
      */
     private static class WrapMethodTransformer extends MethodVisitor {
-        public WrapMethodTransformer(MethodVisitor mv) {
+        private final boolean strictDoubleWrappingCheck;
+
+        public WrapMethodTransformer(MethodVisitor mv, boolean strictDoubleWrappingCheck) {
             super(Opcodes.API_VERSION, mv);
+            this.strictDoubleWrappingCheck = strictDoubleWrappingCheck;
         }
 
         @Override
         public void visitCode() {
             super.visitCode();
 
             mv.visitVarInsn(Opcodes.ALOAD, 0);
-            mv.visitMethodInsn(Opcodes.INVOKESTATIC, "kotlinx/coroutines/flow/internal/FlowValueWrapperInternalKt", "wrapInternalDebuggerCapture", "(Ljava/lang/Object;)Ljava/lang/Object;", false);
+            String prefix;
+            if (strictDoubleWrappingCheck) {
+                prefix = "Strict";
```

Comment:
strictly speaking it is a postfix, but to me looks better to have one method with a parameter

Reply:
for me it is having extra two methods in the library for the debugging purposes only instead of one